### ISSUE TITLE
Input layer dimensions are not set if there is no training data reader

### ISSUE THREAD

**Issue by timmoon10** [LLNL/lbann#1098]
I'd like to run models without a training set. In particular, unit tests with the "check metric" and "check gradient" callbacks only need the test set. However, if I don't specify a training data reader, the input layer is initialized with output dimensions of 0. I can workaround this for now by setting up a dummy data reader for training and setting the number of epochs to zero.

**Comment by samadejacobs** [comment]
I think this is a problem with "inference-only LBANN" too.
